Changelog.md

```diff
@@ -1,5 +1,9 @@
 ### 0.8.21 (unreleased)
 
+Important Bugfixes:
+ * Yul Optimizer: Fix ``Fullinliner`` step (``i``) not preserving the evaluation order of arguments passed into inlined functions when not preceded by ``ExpressionSplitter`` (``x``).
```

Comment:
I'm assuming we're not mentioning the case where function arguments are complex expressions (i.e. not identifiers or literals) since that was how it was supposed to work anyway?

Reply:
All 3 cases were affected (calls, identifiers and literals), but only the first one could result in wrong bytecode. Whether the latter two worked is a bit of a philosophical distinction :) I did not go into detail here simply to keep this shorter. The inliner was not preserving the order and now it does so it's accurate. The bug list entry and the blog post can go into more detail.

Reply:
Yes, I'm aware of that; what I was asking about was the fact that we used to (and still do before this is merged) inline function calls with function call arguments, even though this was never supposed to be the case as per the docs. The major issue here of course was the evaluation order of course.

Reply:
Ah, that's what you meant. Well technically it's a separate bug so I guess it could have a separate entry. Added one now.